### PR TITLE
added template snippets import 

### PR DESCRIPTION
**Pull request by lpramuk**
Updated 'robottelo.cli.import_.Import', Updated 'tests.foreman.cli.test_import.TestImport'

+ added new methods for handling import of template snippets
+ added default positive test exercising import of template snippets

### CODE REVIEW

**Comment by lpramuk** [comment]
```
$ nosetests -m test_import_snippets_default tests/foreman/cli/test_import.py
.
----------------------------------------------------------------------
Ran 1 test in 51.386s

OK
```

File: tests/foreman/cli/test_import.py [hunk 1]
```diff
@@ -1338,3 +1362,41 @@ def test_reimport_enable_rh_repos_negative(self, test_data):
...
+
+        # get the sat6 mapping of the imported organizations
+        imp_orgs = get_sat6_id(
+            Import.csv_to_dataset([files['users']]),
+            import_org[1]
+        )
+        # now to check whether all repos from csv appeared in satellite
+        for imp_org in imp_orgs:
+            self.assertNotEqual(
+                Template.list({'organization-id': imp_org['sat6']}).stdout,
```
Comment (abalakh) [inline tests/foreman/cli/test_import.py]: As per #2840 , please remove `.stdout` part:
```python
Template.list({'organization-id': imp_org['sat6']})
```

File: tests/foreman/cli/test_import.py [hunk 2]
```diff
@@ -1338,3 +1362,41 @@ def test_reimport_enable_rh_repos_negative(self, test_data):
                 ssh_import_org[1], files['channels']
             )
         )
+
+    @data(*gen_import_snippet_data())
```
Comment (sthirugn) [inline tests/foreman/cli/test_import.py]: Please use subTest()

**Comment by sthirugn** [comment]
ACK pending open comments.

**Comment by lpramuk** [comment]
comments addressed
- subtest instead of ddt
- return is treated like output, checking for return code removed

**Comment by lpramuk** [comment]
```$ nosetests -vs -m test_import_snippets_default tests/foreman/cli/test_import.py
...
Ran 1 test in 25.130s

OK```

File: tests/foreman/cli/test_import.py [hunk 3]
```diff
@@ -399,6 +423,9 @@ def tearDown(self):
             u'rm -rf "${HOME}"/.transition_data "${HOME}"/puppet_work_dir'
         )
 
+    def runTest(self):
```
Comment (abalakh) [inline tests/foreman/cli/test_import.py]: Could you please explain the purpose of this func? I can't see any calls for this in your tests. I suppose you override some unittest func then? If so, why don't you call `super`?

**Comment by abalakh** [comment]
ACK pending question

File: tests/foreman/cli/test_import.py (hunk 3 above)
Comment (abalakh) [inline tests/foreman/cli/test_import.py]: OK, googled a little, still don't get it. So `runTest()` is a `unittest` func to actually execute one and the only test in class. So, by overriding it here, you're telling `unittest` to run only `test_import_snippets_default()` from entire `TestImport` class? Isn't it something left from debugging?

**Comment by lpramuk** [comment]
@abalakh Oops, forgot to remove...(I was trying something)
Removed now.

File: robottelo/cli/import_.py [hunk 4]
```diff
@@ -499,3 +499,23 @@ def repository_enable_with_tr_data(cls, options=None):
...
             ]
         return(result, transition_data)
+
+    @classmethod
+    def template_snippet_with_tr_data(cls, options=None):
+        """Import template snippets (from spacewalk-report ks-scripts).
+
+        :returns: A tuple of SSHCommandResult and a List containing the
+            transition data of the Import
+        :raises: AssertionError if a non-zero return code is encountered.
```
Comment (rplevka) [inline robottelo/cli/import_.py]: remove this line, as we no longer raise the `AssertionError`

**Comment by rplevka** [comment]
post ACK :D

**Pull request merged**